fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/mixin/networking/ServerConfigurationNetworkHandlerMixin.java

```diff
@@ -51,14 +85,59 @@ private void initAddon(CallbackInfo ci) {
 		this.addon.lateInit();
 	}
 
-	@Inject(method = "sendConfigurations", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/MinecraftServer;getCombinedDynamicRegistries()Lnet/minecraft/registry/CombinedDynamicRegistries;"))
+	@Inject(method = "sendConfigurations", at = @At("HEAD"), cancellable = true)
 	private void onClientReady(CallbackInfo ci) {
-		this.addon.onClientReady();
+		// Send the initial channel registration packet
+		if (this.addon.startConfiguration()) {
+			assert currentTask == null;
+			ci.cancel();
+			return;
+		}
+
+		// Ready to start sending packets
+		if (!sentConfiguration) {
+			// TODO are we happy for mods to send packets here? Maybe have 2 events
```

Comment:
TODO: I think im going to split this into two events, one for the early tasks and packets, and then the default one after vanilla.